src/highdicom/sr/templates.py

```diff
@@ -1935,20 +2983,427 @@ def __init__(
+            context = ObserverContext(
+                observer_type=item.value,
+                observer_identifying_attributes=attributes
+            )
+            observer_contexts.append(context)
+        return observer_contexts
+
+    def get_subject_contexts(
+        self,
+        subject_class: Optional[CodedConcept] = None
+    ) -> List[SubjectContext]:
+        """Get subject contexts.
+
+        Parameters
+        ----------
+        subject_class: Union[highdicom.sr.coding.CodedConcept, pydicom.sr.coding.Code], optional
+            Type of subject ("Specimen", "Fetus", or "Device") for which should
+            be filtered
+
+        Returns
+        -------
+        List[highdicom.sr.templates.SubjectContext]
+           Subject contexts
+
+        """  # noqa
+        root_item = self[0]
+        matches = [
+            (i + 1, item) for i, item in enumerate(root_item.ContentSequence)
+            if item.name == codes.DCM.SubjectClass
+        ]
+        subject_contexts = []
+        for i, (index, item) in enumerate(matches):
+            if subject_class is not None:
+                if item.value != subject_class:
+                    continue
+            try:
+                next_index = matches[i + 1][0]
+            except IndexError:
+                next_index = -1
+            if item.value == codes.DCM.Specimen:
+                attributes = SubjectContextSpecimen.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Fetus:
+                attributes = SubjectContextFetus.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Device:
+                attributes = SubjectContextDevice(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            else:
+                raise ValueError('Unexpected subject class "{item.meaning}".')
+            context = SubjectContext(
+                subject_class=item.value,
+                subject_class_specific_context=attributes
+            )
+            subject_contexts.append(context)
+        return subject_contexts
+
+    def get_planar_roi_measurements(
+        self,
+        tracking_uid: Optional[str] = None,
+        finding_type: Optional[Union[CodedConcept, Code]] = None,
+        finding_site: Optional[Union[CodedConcept, Code]] = None,
+        graphic_type: Optional[
+            Union[GraphicTypeValues, GraphicTypeValues3D]
+        ] = None
+    ) -> List[PlanarROIMeasurementsAndQualitativeEvaluations]:
+        """Get imaging measurements of planar image regions of interest.
+
+        Finds (and optionally filters) content items contained in the
+        CONTAINER content item "Measurement group" as specified by TID 1410
+        "Planar ROI Measurements and Qualitative Evaluations".
```

Comment:
I think this should make it clearer that this returns measurement groups rather than measurements themselves. (same for volumetric and image measurements)

Reply:
After renaming the methods, do you think this would still need further clarification?

Reply:
I would still rephrase slightly to avoid confusion 
```suggestion
        """Get imaging measurement groups of planar image regions of interest.

        Finds (and optionally filters) content items contained in the
        CONTAINER content item "Measurement group" as specified by TID 1410
        "Planar ROI Measurements and Qualitative Evaluations".
```